### ISSUE TITLE
Add support for single files

### ISSUE THREAD

**Issue by ranolfi** [eamodio/vscode-gitlens#321]
Followup of #317 (comments).

Currently, GitLens features are only available when a directory is opened within the editor.

It would be useful if it'd work for single files as well, like the native Git integration in VS Code.

**Comment by corydwood** [comment]
I'll add that this was already working with a previous version of GitLens/VS Code. I don't remember which version though.

**Comment by eamodio** [comment]
I *might* have lied -- dug into it more and it looks like it was pretty easy to get working again (oops). I'd need to give it more burn in before I'd be ready to release it. Would any of you be willing to try out a private build for it?

**Comment by ranolfi** [comment]
@eamodio I'm willing to try it out.

**Comment by eamodio** [comment]
Can you try this private build and see if it resolves the issue for you: 
[gitlens-8.3.3-beta.vsix.zip](https://github.com/eamodio/vscode-gitlens/files/2041202/gitlens-8.3.3-beta.vsix.zip)

To install it, remove the `.zip` from the filename (needed that to make GitHub attach it), and then in the Extensions view click the `...` menu and choose *Install from VSIX...*

Thanks!

**Comment by ranolfi** [comment]
Looking alright so far.

Although I noticed that, upon closing the file in VS Code, the repository will remain active in the "GitLens" section under "Source Control" view in the activity bar, until another file is open.

But I also noticed the same is true for the native Git integration, anyway.

**Comment by eamodio** [comment]
Yeah, it gets a bit tricky to reliably track when the repository should close (if you open multiple files, etc). I have added support to "close" a repository which won't completely remove it, will hide it from the *GitLens* explorer from then on.

I have also discovered that vscode doesn't provide a way to watch for file changes outside the folder(s) opened, so if you open single files, the repository for it doesn't update properly to reflect changes.

The work to fix that is non-trivial and nothing I want to tackle at this point, but if https://github.com/Microsoft/vscode/issues/3025 is added, then I can easily add that support.
